google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -394,8 +394,11 @@ protected static String getAppEngineProjectId() {
       String serviceAccountName = (String) method.invoke(appIdentityService);
       int indexOfAtSign = serviceAccountName.indexOf('@');
       return serviceAccountName.substring(0, indexOfAtSign);
-    } catch (Exception ignore) {
-      // return null if can't determine
+    } catch (Exception exception) {
+      if (System.getProperty("com.google.appengine.runtime.version") != null) {
+        // Could not resolve appengine classes under GAE environment.
+        throw new RuntimeException("Unable to resolve appengine classes. Missing dependency.");
```

Comment:
Maybe we could provide a link to docs somewhere for more information? Maybe in a README file somewhere?